test/integration_test/kubevirt_hyperconv_test.go

```diff
@@ -244,25 +309,197 @@ func verifyInitialVMI(t *testing.T, testState *kubevirtTestState) {
+			Name: dvName,
+		},
+		Spec: cdiv1beta1.DataVolumeSpec{
+			Source: &cdiv1beta1.DataVolumeSource{
+				Blank: &cdiv1beta1.DataVolumeBlankImage{},
+			},
+			Storage: &cdiv1beta1.StorageSpec{
+				StorageClassName: &testState.vmDisks[0].storageClassName,
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteMany,
+				},
+				VolumeMode: &volumeMode,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceName(corev1.ResourceStorage): resource.MustParse("3Gi"),
+					},
+				},
+			},
+		},
+	}
+	if wantOwnerRefOnDV {
+		dv.ObjectMeta.OwnerReferences = []metav1.OwnerReference{
+			{
+				APIVersion: "kubevirt.io/v1",
+				Kind:       "VirtualMachine",
+				Name:       testState.vmiName, // VM name is the same as VMI name
+				UID:        types.UID(testState.vmUID),
+			},
+		}
+	}
+	log.Infof("Creating a hotplug data volume %s/%s", ns, dvName)
+	_, err := kvCli.CdiClient().CdiV1beta1().DataVolumes(ns).Create(ctx, dv, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	// add hotplug volume to the VMI
+	opts := &kubevirtv1.AddVolumeOptions{
+		Name: dvName,
+		Disk: &kubevirtv1.Disk{
+			DiskDevice: kubevirtv1.DiskDevice{
+				Disk: &kubevirtv1.DiskTarget{Bus: kubevirtv1.DiskBusSCSI},
+			},
+		},
+		VolumeSource: &kubevirtv1.HotplugVolumeSource{
+			DataVolume: &kubevirtv1.DataVolumeSource{
+				Name: dvName,
+			},
+		},
+	}
+	log.Infof("Adding hotplug data volume %s/%s to VMI %s", ns, dvName, testState.vmiName)
+	err = kvCli.VirtualMachineInstance(ns).AddVolume(ctx, testState.vmiName, opts)
+	require.NoError(t, err)
+
+	// wait until the PVC is bound
+	var pvc *corev1.PersistentVolumeClaim
+	require.Eventuallyf(t, func() bool {
+		pvc, err = core.Instance().GetPersistentVolumeClaim(dvName, ns)
+		if err != nil {
+			log.Warnf("Failed to get PVC for DataVolume %s/%s of context %s: %v",
+				ns, appCtx.App.NameSpace, appCtx.App.Key, err)
+			return false
+		}
+		if pvc.Status.Phase != corev1.ClaimBound {
+			log.Warnf("Waiting for PVC %s/%s phase to be %s; current value: %s",
+				pvc.Namespace, pvc.Name, corev1.ClaimBound, pvc.Status.Phase)
+			return false
+		}
+		log.Infof("PVC %s/%s phase is %s", pvc.Namespace, pvc.Name, pvc.Status.Phase)
+		return true
+	}, 3*time.Minute, 5*time.Second, "PVC %s/%s did not become bound", ns, dvName)
+
+	hpDisk := &hotPlugDisk{
+		pvcName:          pvc.Name,
+		pvName:           pvc.Spec.VolumeName,
+		storageClassName: *pvc.Spec.StorageClassName,
+	}
+	hpDisk.apiVol, err = volumeDriver.InspectVolume(hpDisk.pvName)
+	require.NoError(t, err, "Failed to inspect PV %s for %s", hpDisk.pvName, hpDisk)
+	return hpDisk
+}
+
+func verifyHotPlugDisk(t *testing.T, testState *kubevirtTestState, hpDisk *hotPlugDisk, waitForVPSFixJob bool) {
+	vmDisk := testState.vmDisks[0]
+	vmDiskReplicas := getReplicaNodeIDs(vmDisk.apiVol)
+	vmDiskLabelVal := vmDisk.apiVol.Spec.VolumeLabels[vpsVolAffinityLabel]
+
+	isCollocatedFn := func() bool {
```

Comment:
Can we move this to it's own standalone function? It feels as if this will be useful in other tests too.